tests/manage/pv_services/test_resource_deletion_during_pvc_pod_creation_and_io.py

```diff
@@ -337,6 +371,10 @@ def test_resource_deletion_during_pvc_pod_creation_and_io(
             f"{final_pods_num}"
         )
 
+        if switch_to_provider_needed:
```

Comment:
Is the condition here correct?
We are switching to consumer in next line

Reply:
Yes because if switch_to_provider_needed is True the current context is provider because switching is already done in line 363. Now next step need consumer. So switching to consumer now.

Reply:
ack